applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -290,24 +295,52 @@ namespace MPMParticleGeneratorUtility
 
-                        Matrix shape_functions_values;
+                        // Check number of particles per condition to be created
+                        bool is_equal_int_volumes = false; // default GAUSS
+                        if (i->Has( IS_EQUAL_DISTRIBUTED )){
+                            is_equal_int_volumes = i->GetValue(IS_EQUAL_DISTRIBUTED);
+                        }
 
-                        // Get integration method and shape function values
-                        IntegrationMethod int_method = GeometryData::IntegrationMethod::GI_GAUSS_1;
-                        bool is_equal_int_volumes = false;
+                        std::vector<IntegrationPoint<3>> integration_points;
+                        IndexType number_of_points_per_span;
+                        array_1d<double, 3> local_coordinates;
 
-                        DetermineConditionIntegrationMethodAndShapeFunctionValues(r_geometry, particles_per_condition,
-                            int_method, shape_functions_values, is_equal_int_volumes);
+                        const GeometryData::KratosGeometryType geo_type = r_geometry.GetGeometryType();
+                        IntegrationMethod integration_method = r_geometry.GetDefaultIntegrationMethod();
 
-                        // Number of integration point per condition
-                        const unsigned int integration_point_per_conditions = shape_functions_values.size1();
-                        Vector int_volumes (integration_point_per_conditions);
+                        if (is_equal_int_volumes){
+                            if (geo_type == GeometryData::KratosGeometryType::Kratos_Line2D2  || geo_type == GeometryData::KratosGeometryType::Kratos_Line3D2)
+                            {
+                                number_of_points_per_span = particles_per_condition;
+                                std::vector<double> spans = {-1, 1};
+                                
+                                auto integration_info = IntegrationInfo(r_geometry.LocalSpaceDimension(), number_of_points_per_span, IntegrationInfo::QuadratureMethod::GRID);
+                                
+                                IntegrationPointUtilities::CreateIntegrationPoints1D(
+                                            integration_points, spans, integration_info);
 
-                        if (is_equal_int_volumes) {
-                            for (size_t j = 0; j < integration_point_per_conditions; ++j)  int_volumes[j] = r_geometry.Area() / integration_point_per_conditions;
+                                integration_method = integration_info.GetIntegrationMethod(0);
+                            }
+                            else{
+                                std::string warning_msg = "Equal distribution of particle conditions only available for line segments: ";
+                                KRATOS_INFO("MPMParticleGeneratorUtility") << "WARNING: " << warning_msg << std::endl;
```

Comment:
If this is a warning maybe we should use `KRATOS_WARNING` (this is not the only line where `KRATOS_INFO` is used to display a warning in this file)

Reply:
I changed it in the particle generator utility as suggested.